### PR TITLE
[RFC] Save boot params in C function

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commit
"core: ffa: parse boot info from SPMC at EL3".

`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commits
"core: arm: refactor boot argument handling" and
"core: remove fdt argument from get_aslr_seed()".

For commit "core: remove unused arguments from boot_init_primary_early()":
Update also core/arch/riscv/kernel/boot.c.

File: core/arch/arm/kernel/entry_a32.S [hunk 1]
```diff
@@ -633,10 +600,10 @@ shadow_stack_access_ok:
 	/* Pass the vector address returned from main_init */
 	sub	r1, r1, r0
 #else
-	/* realy standard bootarg #1 and #2 to non secure entry */
+	/* relay standard bootarg #1 and #2 to non secure entry */
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/kernel/entry_a32.S:603]: Thanks, I'll fix.

**Comment by jenswi-linaro** [comment]
Tags applied, comments addressed.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commit
"core: remove unused arguments from boot_init_primary_early()".

**Comment by jenswi-linaro** [comment]
Squashed and tags applied.

**Comment by jforissier** [comment]
@raymo200915 would you like to give any Acked-by: / Reviewed-by: tag for any of these commits?

**Comment by etienne-lms** [comment]
@jforissier, actually @raymo200915 is waiting for this P-R to be merged: https://github.com/OP-TEE/optee_os/pull/6308#issuecomment-1761552724.

**Comment by jforissier** [comment]
> @jforissier, actually @raymo200915 is waiting for this P-R to be merged: [#6308 (comment)](https://github.com/OP-TEE/optee_os/pull/6308#issuecomment-1761552724).

Yes I know that, but still he doesn't seem to want to Ack here :wink: no worries.

**Comment by raymo200915** [comment]
> > @jforissier, actually @raymo200915 is waiting for this P-R to be merged: [#6308 (comment)](https://github.com/OP-TEE/optee_os/pull/6308#issuecomment-1761552724).
> 
> Yes I know that, but still he doesn't seem to want to Ack here 😉 no worries.

Sorry for the late, add `Acked-by: Raymond Mao <raymond.mao@linaro.org>`.

**Pull request merged**